SabreTools.Serialization/Wrappers/PortableExecutable.Extraction.cs

```diff
@@ -158,6 +160,88 @@ public bool ExtractCExe(string outputDirectory, bool includeDebug)
                 return false;
             }
         }
+        
+        /// <summary>
+        /// Extract data from an InstallShield Executable
+        /// </summary>
+        /// <param name="outputDirectory">Output directory to write to</param>
+        /// <param name="includeDebug">True to include debug data, false otherwise</param>
+        /// <returns>True if extraction succeeded, false otherwise</returns>
+        public bool ExtractInstallShieldExecutable(string outputDirectory, bool includeDebug)
+        {
+            try
+            {
+                var overlayAddress = (int)OverlayAddress;
+                
+                // Return if overlay doesn't exist.
+                if (overlayAddress == -1) 
+                    return false;
+                
+                // Ensure the stream is starting at the overlay address
+                _dataSource.Seek(overlayAddress, SeekOrigin.Begin);
+
+                var streamLength = _dataSource.Length;
+                const int chunkSize = 65536;
+
+                while (_dataSource.Position < streamLength)
+                {
+                    lock (_dataSourceLock)
+                    {
+                        // Try to deserialize the source data
+                        var deserializer = new Readers.InstallShieldExecutableFile();
+                        var entry = deserializer.Deserialize(_dataSource);
+                        if (entry == null)
+                            return false;
+                    
+                        // Get the length, and make sure it won't EOF
+                        var length = entry.Length;
```

Comment:
`uint` can be cast to `long` to avoid any weird math down below. it also ensures that your `length > 0` check will actually succeed if somehow it overreads.